homeassistant/components/lacrosse_view/__init__.py

```diff
@@ -17,41 +18,49 @@
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
 
+@dataclass
+class LaCrosseCoordinatorData:
+    """Data for the get_data function."""
+
+    last_update: datetime
+    api: LaCrosse
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up LaCrosse View from a config entry."""
 
     async def get_data() -> list[Sensor]:
         """Get the data from the LaCrosse View."""
         now = datetime.utcnow()
 
-        if hass.data[DOMAIN][entry.entry_id]["last_update"] < now - timedelta(
-            minutes=59
-        ):  # Get new token
-            hass.data[DOMAIN][entry.entry_id]["last_update"] = now
-            await api.login(entry.data["username"], entry.data["password"])
+        if data.last_update < now - timedelta(minutes=59):  # Get new token
```

Comment:
Accessing this variable set after the function is defined in the scope of the outer function seems error prone. It might be simpler to inherit from the data update coordinator and store the LaCrosseCoordinatorData fields as members.

Reply:
Yeah, extracting a custom coordinator class is even better.